cloudinit/config/cc_growpart.py

```diff
@@ -93,6 +95,55 @@ class RESIZE(object):
 LOG = logging.getLogger(__name__)
 
 
+@lru_cache()
+def is_lvm_lv(devpath):
+    if util.is_Linux():
+        # all lvm lvs will have a realpath as a 'dm-*' name.
+        rpath = os.path.realpath(devpath)
+        if not os.path.basename(rpath).startswith("dm-"):
+            return False
+        out, _ = subp.subp("udevadm", "info", devpath)
+        # lvs should have DM_LV_NAME=<lvmuuid> and also DM_VG_NAME
+        return 'DM_LV_NAME=' in out
+    else:
+        return False
+
+
+@lru_cache()
+def get_pvs_for_lv(devpath):
+    myenv = {'LANG': 'C'}
+
+    if util.is_Linux():
+        try:
+            (out, _err) = subp.subp(["lvm", "lvs", devpath, "--options=vgname",
+                                     "--noheadings"], update_env=myenv)
+            vgname = out.strip()
+        except subp.ProcessExecutionError as e:
+            if e.exit_code != 0:
+                util.logexc(LOG, "Failed: can't get Volume Group information "
+                            "from %s", devpath)
+                raise ResizeFailedException(e) from e
+
+        try:
+            (out, _err) = subp.subp(["lvm", "vgs", vgname, "--options=pvname",
+                                     "--noheadings"], update_env=myenv)
+            pvs = [p.strip() for p in out.splitlines()]
+            if len(pvs) > 1:
+                LOG.info("Do not know how to resize multiple Physical"
+                         " Volumes")
+            else:
+                return pvs[0]
+        except subp.ProcessExecutionError as e:
+            if e.exit_code != 0:
+                util.logexc(LOG, "Failed: can't get Physical Volume "
+                            "information from Volume Group %s", vgname)
+                raise ResizeFailedException(e) from e
+    else:
+        LOG.warning("No support for get_pvs_for_lv on %s",
+                    platform.system())
```

Comment:
I really do not want a warning here.  Doing so means that we have a warning in cases that previously did not.  Someone had an image that was working perfectly fine for them (two pvs for /).  It just was not resizing anything.  But now they'll have a warning in their logs, and in cloud-init speak, warnings are pretty much errors.